modules/market/src/main/java/com/opengamma/strata/market/key/QuotesArrayKey.java

```diff
@@ -0,0 +1,420 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * <p>
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.key;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.market.FieldName;
+import com.opengamma.strata.basics.market.MarketDataBox;
+import com.opengamma.strata.basics.market.ScenarioMarketDataKey;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.id.StandardId;
+import com.opengamma.strata.market.value.QuotesArray;
+
+/**
+ * A key identifying a {@link QuotesArray} containing values for a piece of quoted market data in multiple scenarios.
+ */
+@BeanDefinition
+public final class QuotesArrayKey
+    implements ScenarioMarketDataKey<Double, QuotesArray>, ImmutableBean {
+
+  /**
+   * The ID of the market data that is required, typically an ID from an external data provider.
+   * For example, 'Bloomberg~AAPL'.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final StandardId standardId;
+
+  /**
+   * The field name in the market data record that is required.
+   * For example, {@link FieldName#MARKET_VALUE}.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final FieldName fieldName;
```

Comment:
Think it would make more sense if it held a `QuoteKey` instead of the two parts of a quote key. After all, its the same concept, just a wrapper for a different storage.